helm/dagger/values.yaml

```diff
@@ -88,9 +110,74 @@ engine:
     create: false
     annotations: []
 
-  existingServiceAccount: {}
+  existingServiceAccount:
+    {}
     # name: "default"
 
+  ### Configure lifecycle hooks for the Dagger Engine
+  #
+  lifecycle:
+    # PreStop hook executed before container termination
+    preStop:
+      # exec:
+      #   command: ["/bin/sh", "-c", "sleep 10"]  # Example: graceful shutdown delay
+
+  ### Storage configuration for StatefulSet
```

Comment:
I like that you captured the `StatefulSet` applicability @jholm117 ! WDYT about nesting this entire config under `statefulSet`?

Reply:
I moved the pvc template config under statefulSet, but I kept the hostpath config outside that scope since it applies to the daemonset as well (and renamed the keys). Let me know what you think!